gravitee-apim-plugin/gravitee-apim-plugin-entrypoint/gravitee-apim-plugin-entrypoint-http-get/src/main/java/io/gravitee/plugin/entrypoint/http/get/HttpGetEntrypointConnector.java

```diff
@@ -0,0 +1,310 @@
+import io.reactivex.Flowable;
+import io.vertx.core.json.JsonObject;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicBoolean;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * @author Florent CHAMFROY (florent.chamfroy at graviteesource.com)
+ * @author GraviteeSource Team
+ */
+@AllArgsConstructor
+@Slf4j
+public class HttpGetEntrypointConnector implements EntrypointAsyncConnector {
+
+    /**
+     * Internal attribute used to store the id of the last message sent to a client. Restricted for this entrypoint.
+     */
+    static final String ATTR_INTERNAL_LAST_MESSAGE_ID = "last.message.id";
+    static final String CURSOR_QUERY_PARAM = "cursor";
+    static final String LIMIT_QUERY_PARAM = "limit";
+
+    static final Set<ConnectorMode> SUPPORTED_MODES = Set.of(ConnectorMode.SUBSCRIBE);
+
+    protected final HttpGetEntrypointConnectorConfiguration configuration;
+
+    @Override
+    public ListenerType supportedListenerType() {
+        return ListenerType.HTTP;
+    }
+
+    @Override
+    public Set<ConnectorMode> supportedModes() {
+        return SUPPORTED_MODES;
+    }
+
+    @Override
+    public int matchCriteriaCount() {
+        return 1;
+    }
+
+    @Override
+    public boolean matches(final ExecutionContext ctx) {
+        return (HttpMethod.GET == ctx.request().method());
+    }
+
+    @Override
+    public Completable handleRequest(final ExecutionContext ctx) {
+        return Completable.fromRunnable(
+            () -> {
+                ctx.putInternalAttribute(
+                    InternalContextAttributes.ATTR_INTERNAL_MESSAGES_LIMIT_DURATION_MS,
+                    configuration.getMessagesLimitDurationMs()
+                );
+
+                int messagesLimitCount = configuration.getMessagesLimitCount();
+                if (ctx.request().parameters().containsKey(LIMIT_QUERY_PARAM)) {
+                    String limit = ctx.request().parameters().getFirst(LIMIT_QUERY_PARAM);
+                    if (limit != null && !limit.isEmpty()) {
+                        messagesLimitCount = Math.min(messagesLimitCount, Integer.parseInt(limit));
+                    }
+                }
+                ctx.putInternalAttribute(InternalContextAttributes.ATTR_INTERNAL_MESSAGES_LIMIT_COUNT, messagesLimitCount);
+
+                if (ctx.request().parameters().containsKey(CURSOR_QUERY_PARAM)) {
+                    String cursor = ctx.request().parameters().getFirst(CURSOR_QUERY_PARAM);
+                    if (cursor != null && !cursor.isEmpty()) {
+                        ctx.putInternalAttribute(InternalContextAttributes.ATTR_INTERNAL_MESSAGES_RESUME_LASTID, cursor);
+                    }
+                }
+            }
+        );
+    }
+
+    @Override
+    public Completable handleResponse(final ExecutionContext ctx) {
+        String acceptHeader = ctx.request().headers().get(HttpHeaderNames.ACCEPT);
```

Comment:
Yuu should wrap this code with Completable.fromRunnable otherwise it will be executed while building chain call instead of at subscription time.